### PR TITLE
chore(dot/digest): add digest to `loglevel` config

### LINKED ISSUE #1851: config: add `dot/digest` to package log level config
## Task summary

- add a config option for log level for the `digest` package to the config file
- pass this into `digest.NewHandler` and use to initialize the package logger

### PR DESCRIPTION
**Pull request by 1garo**
## Changes

- Add digest to config pkg
- Pass log as a parameter on `digest.newHandle`
- Add test for `createDigestHandler`

## Tests

```
go test ./dot/digest
```

## Issues

- closes #1851 

## Primary Reviewer

- @noot

### CODE REVIEW

**Comment by 1garo** [comment]
I was having some problems while trying to run `go test ./dot/services_test`, it wasn't finding the path to `.../gssmr/genesis.json`, the problem was at:

image

Talking to @EclesioMeloJunior, he told be that was already fixing it in his feature and get the path dynamically.

just FYI :)

File: dot/digest/digest.go [hunk 1]
```diff
@@ -41,6 +40,9 @@ type Handler struct {
 	grandpaForcedChange    *grandpaChange
 	grandpaPause           *pause
 	grandpaResume          *resume
+
+	// log
```
Comment (qdm12) [inline dot/digest/digest.go:44]:
```suggestion
```

File: dot/digest/digest.go [hunk 2]
```diff
@@ -57,7 +59,8 @@ type resume struct {
 }
 
 // NewHandler returns a new Handler
-func NewHandler(blockState BlockState, epochState EpochState, grandpaState GrandpaState) (*Handler, error) {
+func NewHandler(blockState BlockState, epochState EpochState, grandpaState GrandpaState,
+	logger log.LeveledLogger) (*Handler, error) {
```
Comment (qdm12) [inline dot/digest/digest.go:63]: :100: for the dependency injection

File: dot/digest/digest_test.go [hunk 3]
```diff
@@ -38,7 +38,8 @@ func newTestHandler(t *testing.T) *Handler {
 	err = stateSrvc.Start()
 	require.NoError(t, err)
 
-	dh, err := NewHandler(stateSrvc.Block, stateSrvc.Epoch, stateSrvc.Grandpa)
+	logger := log.NewFromGlobal(log.AddContext("pkg", "digest"))
```
Comment (qdm12) [inline dot/digest/digest_test.go:41]: Let's get rid of the logs during tests by default :wink: 

```suggestion
	logger := log.New(log.SetWriter(io.Discard)))
```

File: dot/services_test.go [hunk 4]
```diff
@@ -381,3 +384,25 @@ func Test_createPprofService(t *testing.T) {
 
 	require.NotNil(t, service)
 }
+
+func Test_createDigestHandler(t *testing.T) {
+	testDatadirPath := t.TempDir()
+	config := state.Config{
+		Path:     testDatadirPath,
+		LogLevel: log.Info,
```
Comment (qdm12) [inline dot/services_test.go:392]: nit perhaps change that to a higher level like

```suggestion
		LogLevel: log.Warn,
```

So our tests are less log spammed by default? :thinking:

File: dot/services_test.go [hunk 5]
```diff
@@ -381,3 +384,25 @@ func Test_createPprofService(t *testing.T) {
...
+	err := stateSrvc.Initialise(gen, genHeader, genTrie)
+	require.NoError(t, err)
+
+	err = stateSrvc.Start()
+	require.NoError(t, err)
+
+	dh, err := createDigestHandler(stateSrvc)
+	require.NoError(t, err)
+	require.NotNil(t, dh)
+}
```
Comment (qdm12) [inline dot/services_test.go:408]: Awesome, thanks for adding the test here!

**Review by qdm12 — APPROVED** [review]

File: dot/services_test.go (hunk 4 above)
Comment (1garo) [reply dot/services_test.go:392]: Make sense

File: dot/digest/digest_test.go (hunk 3 above)
Comment (noot) [reply dot/digest/digest_test.go:41]: I don't think we need to entirely get rid of them, we might want them at some point, maybe just set to LevelCrit?

**Review by noot — APPROVED** [review]
nice work!

**Comment by 1garo** [comment]
depends on pr below to pass `unit-tests` CI:

- #2091

File: dot/digest/digest_test.go (hunk 3 above)
Comment (qdm12) [reply dot/digest/digest_test.go]: Same thing, you can just change `io.Discard` with `os.Stdout` :wink:

**Pull request merged**